### PR TITLE
`Programming exercises`: Fix an issue when resetting test cases

### PR DESCRIPTION
**Pull request by simon-hng**
#### General

- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/guidelines/development-process/#naming-conventions-for-github-pull-requests).

#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I documented the Java code using JavaDoc style.
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context

Accessing the endpoint `api/programming-exercises/{{exerciseId}}/test-cases/reset` currently does not reset visibility but all other settings. Resetting a test case should also reset its visibility to the default configuration (always) 

### Description

Updates `programmingExerciseTestCaseService.reset(exerciseId)` to also reset visibility.
This method is only used by the above mentioned rest endpoint.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Go to the grading/test case configuration
2. Make some changes to the test cases settings. Do make changes to the visibility so at least one test case does not have visibility == always.
3. Click on the save button and wait for it to finish.
4. Click on the reset button and check that visibility and other options are correctly reset.

### CODE REVIEW

**Review by Strohgelaender — APPROVED** [review]
code looks good.

Would it make sense to add a server test for this case?

**Review by Strohgelaender — DISMISSED** [review]
code looks good

File: src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseTestCaseResource.java [hunk 1]
```diff
@@ -98,7 +98,8 @@ public ResponseEntity<Set<ProgrammingExerciseTestCase>> updateTestCases(@PathVar
     }
 
     /**
-     * Use with care: Set the weight of all test cases of an exercise to 1.
+     * Use with care: Resets all test cases of an exercise to their initial configuration
+     * Set all test case weights to 1, all bonus multipliers to 1 all bonus points to 0 and visibility to always.
```
Comment (bassner) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ProgrammingExerciseTestCaseResource.java:102]:
```suggestion
     * Set all test case weights to 1, all bonus multipliers to 1, all bonus points to 0 and visibility to always.
```

File: src/main/webapp/app/exercises/programming/manage/services/programming-exercise-grading.service.ts [hunk 2]
```diff
@@ -106,9 +106,10 @@ export class ProgrammingExerciseGradingService implements IProgrammingExerciseGr
     }
 
     /**
-     * Use with care: Set all test case weights to 1, all bonus multipliers to 1 and all bonus points to 0.
+     * Use with care: Resets all test cases of an exercise to their initial configuration
+     * Set all test case weights to 1, all bonus multipliers to 1 all bonus points to 0 and visibility to always.
```
Comment (bassner) [inline src/main/webapp/app/exercises/programming/manage/services/programming-exercise-grading.service.ts:110]:
```suggestion
     * Set all test case weights to 1, all bonus multipliers to 1, all bonus points to 0 and visibility to always.
```

**Review by bassner — APPROVED** [review]
LGTM!

**Review by bassner — APPROVED** [review]

**Pull request merged**